response_operations_ui/views/messages.py

```diff
@@ -206,29 +242,37 @@ def view_selected_survey(selected_survey):
         limit = request.args.get('limit', default=10, type=int)
         flash_message = request.args.get('flash_message', default="", type=str)
         conversation_tab = request.args.get('conversation_tab', default='open')
-
-        thread_count = message_controllers.get_conversation_count(
-            {'survey': survey_id,
-             'is_closed': conversation_tab == 'closed',
-             'my_conversations': conversation_tab == 'my messages',
-             'new_respondent_conversations': conversation_tab == 'initial'})
+        ru_ref_filter = request.args.get('ru_ref_filter', default='')
+        business_id_filter = request.args.get('business_id_filter', default='')
+
+        form = SecureMessageRuFilterForm()
+
+        if form.validate_on_submit():
+            new_ru_ref = form.ru_ref_filter.data
+            if new_ru_ref and new_ru_ref != ru_ref_filter:
+                business_id_filter = party_controller.try_get_party_id_by_ru_ref(new_ru_ref)
+                if business_id_filter:
+                    ru_ref_filter = new_ru_ref
+                else:
+                    ru_ref_filter = ''
+                    flash_message = f"Filter not applied: {new_ru_ref} is an unknown RU ref"
```

Comment:
Bit nitpicky but this wording could probably be improved.  Maybe something like `{new_ru_ref} not found in system` is a bit more succinct?